### PR TITLE
mon: have 'mon stat' output json as well

### PR DESCRIPTION
**Pull request by jecluis**
Fixes: https://tracker.ceph.com/issues/46816

Signed-off-by: Joao Eduardo Luis `<joao@suse.com>`

### CODE REVIEW

File: src/mon/MonmapMonitor.cc [hunk 1]
```diff
@@ -305,10 +305,30 @@ bool MonmapMonitor::preprocess_command(MonOpRequestRef op)
...
+      f->close_section();  // quorum
+      f->close_section();  // monmap
+      f->flush(ss);
+    } else {
+      mon->monmap->print_summary(ss);
+      ss << ", election epoch " << mon->get_epoch() << ", leader "
+         << mon->get_leader() << " " << mon->get_leader_name()
+         << ", quorum " << mon->get_quorum()
+         << " " << mon->get_quorum_names();
+		}
```
Comment (tchaikov) [inline src/mon/MonmapMonitor.cc:330]: indent.

File: src/mon/MonmapMonitor.cc [hunk 2]
```diff
@@ -305,10 +305,30 @@ bool MonmapMonitor::preprocess_command(MonOpRequestRef op)
...
-    ss << ", election epoch " << mon->get_epoch() << ", leader "
-       << mon->get_leader() << " " << mon->get_leader_name()
-       << ", quorum " << mon->get_quorum() << " " << mon->get_quorum_names();
+    if (f) {
+      f->open_object_section("monmap");
+      mon->monmap->dump_summary(f.get());
+      f->dump_string("leader", mon->get_leader_name());
+      f->open_array_section("quorum");
+      for (set<int>::iterator rank = mon->get_quorum().begin();
+           rank != mon->get_quorum().end(); ++rank) {
```
Comment (tchaikov) [inline src/mon/MonmapMonitor.cc:314]:
```suggestion
      for (auto rank : mon->get_quorum()) {
        std::string name = mon->monmap->get_name(rank);
        // ...
```

**Review by jan--f — APPROVED** [review]
:+1: for @tchaikov's suggested changes. Otherwise lgtm

**Comment by jecluis** [comment]
pushed changes according to suggestions; added test.

**Review by tchaikov — APPROVED** [review]

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2020-10-17_08:59:57-rados-wip-kefu-testing-2020-10-17-1516-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2020-10-18_16:30:00-rados-wip-kefu-testing-2020-10-18-2239-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2020-10-19_04:31:12-rados-wip-kefu-testing-2020-10-18-2239-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2020-10-19_06:06:20-rados-wip-kefu-testing-2020-10-18-2239-distro-basic-smithi

failures tracked by

- https://tracker.ceph.com/issues/47024

**Comment by neha-ojha** [comment]
This PR is causing https://tracker.ceph.com/issues/47907.

**Pull request merged**